UnityProject/Assets/Scripts/Objects/Doors/DoorMasterController.cs

```diff
@@ -164,6 +164,10 @@ public void UpdateNewPlayer(NetworkConnection playerConn)
 
 		private void TryBump()
 		{
+			if (matrix.GetFirst<FireLock>(registerTile.LocalPositionServer, true) != null)
+			{
+				if(matrix.GetFirst<FireLock>(registerTile.LocalPositionServer, true).fireAlarm.activated) return;
+			}
```

Comment:
I don't trust anything in matrix to be cheap. Best to save this to a variable.
```suggestion
			var firelock = matrix.GetFirst<FireLock>(registerTile.LocalPositionServer, true);
			if (firelock != null && firelock.fireAlarm.activated) return;
```

Though it is interesting the state of the fire alarm determines the behaviour here. Wouldn't any closed firedoor need this?